Sources/CodeScanner/CodeScanner.swift

```diff
@@ -239,6 +243,36 @@ public struct CodeScannerView: UIViewControllerRepresentable {
         override public var supportedInterfaceOrientations: UIInterfaceOrientationMask {
             return .all
         }
+        
+        /** Touch the screen for autofocus */
+        public override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+            // AutoFocus
+            if touches.first?.view == view {
+                let videoView = view
+                
+                let screenSize = videoView!.bounds.size
+                if let touchPoint = touches.first {
+                    let xPoint = touchPoint.location(in: videoView).y / screenSize.height
+                    
+                    let yPoint = 1.0 - touchPoint.location(in: videoView).x / screenSize.width
+                    let focusPoint = CGPoint(x: xPoint, y: yPoint)
+                    
+                    if let device = videoCaptureDevice {
+                        do {
+                            try device.lockForConfiguration()
+                            
+                            // Focus to the correct point, make continiuous focus and exposure so the point stays sharp when moving the device closer
+                            device.focusPointOfInterest = focusPoint
+                            device.focusMode = .continuousAutoFocus
+                            device.exposurePointOfInterest = focusPoint
+                            device.exposureMode = AVCaptureDevice.ExposureMode.continuousAutoExposure
+                            device.unlockForConfiguration()
+                        } catch {
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
I've restructured this function a bit. The code is still the same, I just recommend returning early if possible, hence the guard statement at the beginning. This frees you from having to check for nil a lot in the middle of the function, and allows you to write code that is less heavily indented.

```suggestion
        /** Touch the screen for autofocus */
        public override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
            guard touches.first?.view == view,
                  let touchPoint = touches.first,
                  let device = videoCaptureDevice
            else { return }
            
            let videoView = view
            let screenSize = videoView!.bounds.size
            let xPoint = touchPoint.location(in: videoView).y / screenSize.height
            let yPoint = 1.0 - touchPoint.location(in: videoView).x / screenSize.width
            let focusPoint = CGPoint(x: xPoint, y: yPoint)
            
            do {
                try device.lockForConfiguration()
            } catch {
                return
            }
            
            // Focus to the correct point, make continiuous focus and exposure so the point stays sharp when moving the device closer
            device.focusPointOfInterest = focusPoint
            device.focusMode = .continuousAutoFocus
            device.exposurePointOfInterest = focusPoint
            device.exposureMode = AVCaptureDevice.ExposureMode.continuousAutoExposure
            device.unlockForConfiguration()
        }
```